Reset-Office365Tenant/Reset-Office365Tenant.psm1

```diff
@@ -0,0 +1,163 @@
+# Set strict mode.
+Set-StrictMode -Version 2.0
+
+# Disable progress bar.
+$ProgressPreference = "SilentlyContinue"
+
+# Set default ErrorAction to Stop.
+$ErrorActionPreference = "Stop"
+
+############################################################
+# Add test users to Office 365
+############################################################
+function Add-Office365Users($max, $defaultPassword)
+{
+    # Get available SKUs.
+    $skus = Get-AvailableAccountSku
+
+    # If no SKUs are available
+    if ($false -eq $skus)
+    {
+        return $false
+    }
+    else
+    {
+        # Cap the max number of users created to the number of available SKU
+        $availableSkus = $($skus[0].ActiveUnits - $skus[0].ConsumedUnits)
+        if ($max -gt $availableSkus)
+        {
+            Write-Host -ForegroundColor Yellow "WARNING: Only $($skus[0].ActiveUnits - $skus[0].ConsumedUnits) SKUs for $max."
+            $max = $availableSkus
+        }
+    }
+
+    # Gets the Domains... we'll grab the first one - lazy
+    try
+    {
+        $Domains = Get-MsolDomain
```

Comment:
Get-MsolDomain will work, but as you stated it can return multiple domains.  The cleaner approach would be to use Get-emailaddresspolicy and use the EnabledPrimarySMTPAddressTemplate property to capture the domain.

Reply:
Created a function Get-EnabledPrimarySMTPAddressTemplate and used that instead :)